### PR TITLE
[Yamato] Adding Vulkan to automation pipelines

### PR DESCRIPTION
**Pull request by sophiaaar**
### Purpose of this PR
* All pipelines can now be run in DX11 or Vulkan (except Universal Stereo)
* Adding reference images for Vulkan (hence the high number of changed files)
* Changing name of the OSX pipelines so that it is clear that it is OSX Metal
* Added a script in each project to allow building in different APIs (borrowed from Felipe's work in the Universal test project)

---
### Testing status
Example of Universal Standalone Vulkan working:
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/automation%252Ftest-multiple-apis/.yamato%252Fupm-ci-universal.yml%2523Universal_Windows64_Standalone_Vulkan_trunk/419686/job

---
### Overall Product Risks
**Technical Risk**: None

**Halo Effect**: None

---
### Comments to reviewers
Note that this PR is simply to enable more APIs to be testable in automation, but it does not guarantee that these tests pass.

Testing Vulkan playmode appears to hang after testing has finished. This has been brought up with QA, but I think this PR should still go in as it enables future testing, and Vulkan doesn't run in CI jobs (eg the nightly) anyway.

The failed jobs are because I checked that the jobs would start, and then cancelled them.

### CODE REVIEW

File: .yamato/upm-ci-hdrp.yml [hunk 1]
```diff
@@ -74,7 +83,7 @@ projects:
...
     {% if testplatform.name == "Standalone" %}
     - |
-      ssh -i ~/.ssh/id_rsa_macmini -o "StrictHostKeyChecking=no" bokken@$BOKKEN_DEVICE_IP 'cd ~/ScriptableRenderPipeline/TestProjects/{{ project.folder }} && ~/ScriptableRenderPipeline/TestProjects/{{ project.folder }}/utr/utr {{ testplatform.args }}OSX  --testproject=/Users/bokken/ScriptableRenderPipeline/TestProjects/{{ project.folder }} --editor-location=/Users/bokken/.Editor --artifacts_path=/Users/bokken/ScriptableRenderPipeline/TestProjects/{{ project.folder }}/test-results  --timeout=1200'
+      ssh -i ~/.ssh/id_rsa_macmini -o "StrictHostKeyChecking=no" bokken@$BOKKEN_DEVICE_IP 'cd ~/ScriptableRenderPipeline/TestProjects/{{ project.folder }} && ~/ScriptableRenderPipeline/TestProjects/{{ project.folder }}/utr/utr {{ testplatform.args }}OSX  --testproject=/Users/bokken/ScriptableRenderPipeline/TestProjects/{{ project.folder }} --editor-location=/Users/bokken/.Editor --artifacts_path=/Users/bokken/ScriptableRenderPipeline/TestProjects/{{ project.folder }}/test-results'
```
Comment (phi-lira) [inline .yamato/upm-ci-hdrp.yml]: why are we removing timeout?

File: TestProjects/VisualEffectGraph/Assets/Scripts/Editor/CustomBuild.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+using UnityEditor;
```
Comment (phi-lira) [inline TestProjects/VisualEffectGraph/Assets/Scripts/Editor/CustomBuild.cs]: Adding this script multiple times in each test project is not ideal. You should add once in the GraphisTestFramework.

Comment (sophiaaar) [reply TestProjects/VisualEffectGraph/Assets/Scripts/Editor/CustomBuild.cs]: good idea, thank you

File: .yamato/upm-ci-hdrp.yml (hunk 1 above)
Comment (sophiaaar) [reply .yamato/upm-ci-hdrp.yml]: oh, that wasnt intended - must have happened in a merge. thanks!

Comment (sophiaaar) [reply .yamato/upm-ci-hdrp.yml]: fixed

File: TestProjects/VisualEffectGraph/Assets/Scripts/Editor/CustomBuild.cs (hunk 2 above)
Comment (sophiaaar) [reply TestProjects/VisualEffectGraph/Assets/Scripts/Editor/CustomBuild.cs]: moved the scripts to the graphics test framework, and it looks like this worked

**Pull request merged**